### ISSUE TITLE
Update to latest boot 3 and cloud versions

### ISSUE THREAD

**Issue by ENate** [mohamed-taman/Springy-Store-Microservices#88]
Hi @mohamed-taman  I will offer to update this repository to the latest boot and cloud versions. If that is okay, I will gradually so with your approval. I am using the template for my project and will love to contribute to keep it up to date. Thanks again for this template.

**Comment by mohamed-taman** [comment]
Hi @ENate Thanks for your interest; it would be a great pleasure! I will wait for your PRs and will help review them, and from my side, I will update the JDK version to 21 as well. So we can work together on the upgrade to take it to the next level.

Let's say we need to update to the latest Spring Boot 3.2.1 and the latest Spring Cloud.

**Comment by ENate** [comment]
Yes sure no worries. Thanks and I will like your review because this will be my first PR in Java. Thanks again

**Comment by ENate** [comment]
Sorry, I was not available for long periods in the past weeks. Will hopefully open the PR soon.

**Comment by ENate** [comment]
Hi @mohamed-taman I made changes as discussed but still finding out which deprecated packages can no longer be used based on your initial template. For instance, a number of changes are currently made on how to implement testing when using recent spring cloud stream packages. Do I open a PR first, or post a few questions regarding a number of deprecated spring cloud testing packages? Sorry for the long pause again. Regards.

**Comment by teja2** [comment]
Hi @mohamed-taman can I start working on this issue? I see there is no active PR for this?

**Comment by ENate** [comment]
Hi @teja2 If you want I can open a PR already as promised and you check whether the testing part is well updated. I am already working on these. Some delays made me not to open a PR but I will do so soon.
